risc0/zkvm/src/host/server/exec/tests.rs

```diff
@@ -665,6 +665,17 @@ fn session_limit() {
 
 #[test]
 fn memory_access() {
+    fn session_faulted(session: Result<Session>) -> bool {
+        if cfg!(feature = "enable-fault-proof") {
+            let session = session.unwrap();
+            let last = session.segments.last().unwrap().resolve().unwrap();
+            last.pre_image.compute_id() == FAULT_CHECKER_ID.into()
```

Comment:
There should definitely be an easier way to determine if a session exited with fault.